services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/multipart/S3MultipartClientGetObjectWiremockTest.java

```diff
@@ -0,0 +1,365 @@
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.interceptor.Context;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+
+@WireMockTest
+public class S3MultipartClientGetObjectWiremockTest {
+    public static final String ERROR_CODE = "InternalError";
+    public static final String ERROR_MESSAGE = "We encountered an internal error. Please try again.";
+    public static final String ERROR_BODY = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                                            + "<Error>\n"
+                                            + "  <Code>" + ERROR_CODE + "</Code>\n"
+                                            + "  <Message>" + ERROR_MESSAGE + "</Message>\n"
+                                            + "</Error>";
+    public static final String BUCKET = "Example-Bucket";
+    public static final String KEY = "Key";
+    private static final int MAX_ATTEMPTS = 7;
+    private static final CapturingInterceptor capturingInterceptor = new CapturingInterceptor();
+
+    private S3AsyncClient multipartClient;
+
+    @BeforeEach
+    public void setup(WireMockRuntimeInfo wm) {
+        capturingInterceptor.clear();
+        multipartClient = S3AsyncClient.builder()
+                                       .region(Region.US_EAST_1)
+                                       .endpointOverride(URI.create(wm.getHttpBaseUrl()))
+                                       .multipartEnabled(true)
+                                       .httpClientBuilder(NettyNioAsyncHttpClient.builder().maxConcurrency(100).connectionAcquisitionTimeout(Duration.ofSeconds(100)))
+                                       .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("key", "secret")))
+                                       .overrideConfiguration(
+                                           o -> o.retryStrategy(AwsRetryStrategy.standardRetryStrategy().toBuilder()
+                                                                                .maxAttempts(MAX_ATTEMPTS)
+                                                                                .circuitBreakerEnabled(false)
+                                                                                .build())
+                                               .addExecutionInterceptor(capturingInterceptor))
+                                       .build();
+    }
+
+    @Test
+    public void stub_200_only() {
```

Comment:
Can we follow the test naming convention? It's hard to know what the expected behavior is from the naming https://github.com/aws/aws-sdk-java-v2/blob/f4a194d7e4aa6188cb7d500fea06ccaa9b93b863/.kiro/steering/testing-guidelines.md#test-naming-conventions